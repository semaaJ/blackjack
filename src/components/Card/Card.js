const Card = (props) => {
    const { symbol, colour, rank, classNames } = props;
    const cardCss = "bg-white px-3 shadow-xl ring-1 ring-gray-900/5 w-40 h-60 m-2 ";

    return (
        <div className={classNames ? `${classNames} ${cardCss}` : cardCss}>
            <div className="mx-auto ">
                <div className="flex flex-col w-full">
                    <h1 className={`text-4xl ${ colour }`}>{ symbol }</h1>
                    <h1 className={`text-4xl ${ colour }`}>{ rank }</h1>
                </div>
            
                <div className="flex justify-center">
                    <h1 className={`text-7xl ${ colour }`}>{ rank }</h1>
                </div>
            
                <div className="flex flex-col w-full items-end">
                    <h1 className={`text-4xl ${ colour }`}>{ rank }</h1>
                    <h1 className={`text-4xl ${ colour }`}>{ symbol }</h1>
                </div>
            </div>
        </div>
    )
}

export default Card;