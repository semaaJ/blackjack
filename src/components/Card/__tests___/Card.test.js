import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../Card';


it('should render', () => {
  const tree = renderer.create(<Card />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with text & colour', () => {
    const tree = renderer.create(<Card symbol="test" colour="text-rose-500" rank="test" />).toJSON();
    expect(tree).toMatchSnapshot();
});
