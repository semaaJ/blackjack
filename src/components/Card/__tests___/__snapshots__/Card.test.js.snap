// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="cursor-pointer bg-white px-3 shadow-xl ring-1 ring-gray-900/5 w-40 h-60 m-4"
>
  <div
    className="mx-auto "
  >
    <div
      className="flex flex-col w-full"
    >
      <h1
        className="text-4xl undefined"
      />
      <h1
        className="text-4xl undefined"
      />
    </div>
    <div
      className="flex justify-center"
    >
      <h1
        className="text-7xl undefined"
      />
    </div>
    <div
      className="flex flex-col w-full items-end"
    >
      <h1
        className="text-4xl undefined"
      />
      <h1
        className="text-4xl undefined"
      />
    </div>
  </div>
</div>
`;

exports[`should render with classNames 1`] = `
<div
  className="absolute mt-8 cursor-pointer bg-white px-3 shadow-xl ring-1 ring-gray-900/5 w-40 h-60 m-4"
>
  <div
    className="mx-auto "
  >
    <div
      className="flex flex-col w-full"
    >
      <h1
        className="text-4xl undefined"
      />
      <h1
        className="text-4xl undefined"
      />
    </div>
    <div
      className="flex justify-center"
    >
      <h1
        className="text-7xl undefined"
      />
    </div>
    <div
      className="flex flex-col w-full items-end"
    >
      <h1
        className="text-4xl undefined"
      />
      <h1
        className="text-4xl undefined"
      />
    </div>
  </div>
</div>
`;

exports[`should render with text & colour 1`] = `
<div
  className="cursor-pointer bg-white px-3 shadow-xl ring-1 ring-gray-900/5 w-40 h-60 m-4"
>
  <div
    className="mx-auto "
  >
    <div
      className="flex flex-col w-full"
    >
      <h1
        className="text-4xl text-rose-500"
      >
        test
      </h1>
      <h1
        className="text-4xl text-rose-500"
      >
        test
      </h1>
    </div>
    <div
      className="flex justify-center"
    >
      <h1
        className="text-7xl text-rose-500"
      >
        test
      </h1>
    </div>
    <div
      className="flex flex-col w-full items-end"
    >
      <h1
        className="text-4xl text-rose-500"
      >
        test
      </h1>
      <h1
        className="text-4xl text-rose-500"
      >
        test
      </h1>
    </div>
  </div>
</div>
`;
